### ISSUE TITLE
Skip tests if expected cockpit package is not installed

### ISSUE THREAD

**Issue by jscotka** [cockpit-project/cockpit#8668]
Hi,
I've scheduled upstream tests on donwstream packages, and fro example there is not shipped apps subpackage, so test check-apps fails 
better solution should be to check if expected package is installed, like other tests check that there are expected images downloaded before test starts.
so there should be some @skip decorator with these prerequiesities like
 * qemu images
 * installed packages

it is similar to another issue. I'm not testing kubernetes subpackages, so I've exclueded them installing this subpackage -> would be nice to skip too in case cockpit-kuberneter not installed, instead of fail. Installing of some packages could expect some additional dependencies, what for example are not shipped in base channels. and enabling more channels can hide other issues.
thanks to @martinpitt to find this issue

Otherwise it could be another test, what will check that all  expected packages are build and installed on TEST_OS system installed. test itseld should not fail, but should report some skip, or ok in case "skip" state is not able to use

**Comment by martinpitt** [comment]
A `@skip` decorator with auto-detection is impractical for this, as this cannot be decided from the host test environment. The test needs to start to boot the test VM, then it can check what's installed.

Also, I don't want our upstream tests to skip if a particular cockpit-* package is not installed, as this could easily hide grave bugs such as not packaging them properly for a particular OS, failing to install them, other mis-builds of our test images.

As an immediate solution for your case I recommend deleting test/verify/check-kubernetes and test/verify/check-openshift if you don't install cockpit-kubernetes into your test images. This isn't very pretty, but at least should unblock you.

I think an explicit way of controlling this would be to introduce a new `@skipPackage(pkgname)` decorator into test/common/testlib.py and use that in e. g. check-kubernetes:

```py
@skipImage("No cockpit-kubernetes packaged", "continuous-atomic", "fedora-atomic", "rhel-atomic")
@skipPackage("cockpit-kubernetes")
class TestKubernetes(kubelib.KubernetesCase, kubelib.KubernetesCommonTests):
```
`skipPackage()` would check if `$TEST_SKIP_PACKAGES` contains `pkgname` and auto-generate a humanly readable reason like `Not testing package <pkgname>`.

Then you can set the packages that you don't install in that env var.

Finding all these might take some time, but I suppose we can start with annotating just a few tests for the packages that you need; and your downstream logs should already tell you which tests need to be blacklisted in that way, right?

Would you like to work on a PR for this?

**Comment by jscotka** [comment]
If there is enough time I can take it, but probably I will have time next week.

**Comment by jscotka** [comment]
Should be implemented by PR:
https://github.com/cockpit-project/cockpit/pull/8680/
